Automattic/jetpack
# RNA: `withErrorMessage` HOC

The point of this HOC is to attach error message to other UI components, such as `ActionButton`, `Button`, `TextControl`, etc.
That way we won't duplicate the error messaging code in multiple components, saving time on initial implementation and maintenance.

Partially extracted from #21284 (props to @cbauerman).

#### Changes proposed in this Pull Request:
- create the `withErrorMessage` HOC
- adjust `ActionButton` component to use it
- slightly modify the `ConnectScreen` tags structure

#### Jetpack product discussion
No.

#### Does this pull request change what data or activity we track or use?
1200295095943283-as-1201194602180870

#### Testing instructions:
0. Read [the docs](https://github.com/Automattic/jetpack/pull/21473/files#diff-045f58f760d0e4befff9d88ddba6fe4bcf8faa12330603fa0a127ebca354925c) and confirm that they make sense.
1. Disconnect Jetpack, go to the connection screen.
2. Switch to offline mode (use browser tools, turn off WiFi, or pull out the internet cable) and click "Set up Jetpack".
3. Confirm that the error message appears under the button and looks like the screenshot below.
4. Go back online, and click the button again. Confirm that you see the loading icon instead of its label.
5. Complete the connection flow, confirm that it works well.

#### The error message screenshot:
<img width="426" alt="Screen Shot 2021-10-20 at 11 35 27 AM" src="https://user-images.githubusercontent.com/1341249/138124781-2eb2f37e-c433-4ef9-93c6-04b7ae57b3f7.png">
